x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/ElasticsearchMappings.java

```diff
@@ -362,6 +364,32 @@ public static void addDatafeedConfigFields(XContentBuilder builder) throws IOExc
         .endObject();
     }
 
+    public static void addDataFrameAnalyticsFields(XContentBuilder builder) throws IOException {
+        builder.startObject(DataFrameAnalyticsConfig.ID.getPreferredName())
```

Comment:
You are re-using the ml configuration index? I wonder if there aren't already fields called `id`?

Apart from that: Have we thought about a separate index for data frame analytics? I think we do not need the coupling.

Reply:
This is an important topic. Open questions:

- Should the id be just called `id` for the analytics config?

I wasn't sure. It felt like the simplest thing was to do so. It wouldn't matter if there was another `id` field in the config index as it would also be a `keyword` and thus there would be no conflict. Arguably, the `job` or `datafeed` prefix in `job_id` and `datafeed_id` seems redundant given we have the `config_type` field to differentiate them. I'm not quite sure if I'm missing something else though.

- Should we reuse the same config index for the DF analytics configs?

I would say yes. The number of documents in that config should never really get high enough to put pressure on the single index. On the other hand, having an additional index does increase the shards on the cluster which has its downsides.

What do you think @droberts195 and @davidkyle ?

Reply:
We have decided to call the id just `id`. More discussion needs to be done for the second issue but it shouldn't stall this PR.